docs: rewrite the MCP Inspector documentation for v2 (#3143)
modelcontextprotocol/modelcontextprotocol: docs/docs/2026-07-28/tools/inspector.mdx

| Client  | Invocation                            | What it's for                                                                     |
| ------- | ------------------------------------- | --------------------------------------------------------------------------------- |
| **Web** | `npx @modelcontextprotocol/inspector` | A full graphical inspector in the browser. The default, and the richest surface.  |
| **CLI** | `… --cli`                             | A scriptable, machine-readable client for CI, shell pipelines, and coding agents. |
| **TUI** | `… --tui`                             | An interactive terminal UI, for when a browser isn't available or wanted.         |

All three are built on the same shared core, so a connection behaves identically across them: the same transports, the same configuration files, the same OAuth state on disk, and the same [protocol-era negotiation](/docs/2026-07-28/tools/inspector/protocol-eras).

<Frame caption="The MCP Inspector web client, connected to a server, with the monitoring sidebar pinned so protocol traffic stays visible while you work.">
  <img src="/images/inspector/web-monitor-sidebar.png" />
</Frame>

## Quickstart

> The Inspector requires **Node 22.19.0 or newer** and runs directly through `npx` without installing anything:

Reviewer: ```suggestion
The Inspector requires **Node 22.19.0 or newer** and runs directly through `npx`. No installation is required:
```

Author: Applied in a8e08fe4.

  </Tab>
  <Tab title="CLI">

```bash
# List a server's tools and exit
npx @modelcontextprotocol/inspector --cli node path/to/server/index.js --method tools/list

# Call a tool and pipe the result into jq
npx @modelcontextprotocol/inspector --cli https://api.example.com/mcp --transport http \
  --method tools/call --tool-name get_weather --tool-arg city=Boston --format json | jq .result
```

See [CLI client](/docs/2026-07-28/tools/inspector/cli).

  </Tab>
  <Tab title="TUI">

```bash
npx @modelcontextprotocol/inspector --tui node path/to/server/index.js
```

See [TUI client](/docs/2026-07-28/tools/inspector/tui).

  </Tab>
</Tabs>

### Inspecting published servers

<Tabs>
  <Tab title="npm package">

```bash
npx -y @modelcontextprotocol/inspector npx @modelcontextprotocol/server-filesystem ~/Desktop
```

  </Tab>
  <Tab title="PyPI package">

```bash
npx @modelcontextprotocol/inspector uvx mcp-server-git --repository ~/code/mcp/servers.git
```

  </Tab>
  <Tab title="Remote HTTP server">

```bash
npx @modelcontextprotocol/inspector --server-url https://api.example.com/mcp --transport http
```

  </Tab>
</Tabs>

> Always read a server's own README first — the exact command and arguments it expects are the server's business, and the Inspector simply runs them.

Reviewer: ```suggestion
Always read a server's own README first — every server requires different commands and arguments.
```

Author: Applied in a8e08fe4.
